### PR TITLE
feat: add the new option --memory-meta-storage

### PR DESCRIPTION
**Pull request by confoc**
bootstrap the whole cluster without installing etcd for testing purposes through using the memory storage of metasrv in bare-metal mode.

### CODE REVIEW

**Comment by MichaelScofield** [comment]
related issue: https://github.com/GreptimeTeam/gtctl/issues/174

**Comment by zyy17** [comment]
@confoc The e2e tests are failed. You should fix it.

File: pkg/components/metasrv.go [hunk 1]
```diff
@@ -108,6 +108,7 @@ CHECKER:
 }
 
 func (m *metaSrv) BuildArgs(params ...interface{}) []string {
+
```
Comment (zyy17) [inline pkg/components/metasrv.go]: Those empty lines are not part of the PR, can you discard the changes?

File: cmd/gtctl/cluster_create.go [hunk 2]
```diff
@@ -178,11 +180,15 @@ func NewCluster(args []string, options *clusterCreateCliOptions, l logger.Logger
 	}
 
 	var cluster opt.Operations
+
```
Comment (zyy17) [inline cmd/gtctl/cluster_create.go]: Also, I think the new lines are not part of the PR.

**Comment by zyy17** [comment]
@confoc I'm confused with the PR. I can't see something about the new option of `--memory-meta-storage`?

File: cmd/gtctl/cluster_create.go [hunk 3]
```diff
@@ -178,11 +180,15 @@ func NewCluster(args []string, options *clusterCreateCliOptions, l logger.Logger
...
 	if options.BareMetal {
+
 		l.V(0).Infof("Creating GreptimeDB cluster '%s' on bare-metal", logger.Bold(clusterName))
 
 		var opts []baremetal.Option
 		opts = append(opts, baremetal.WithEnableCache(options.EnableCache))
+		opts = append(opts, baremetal.WithEnableEtcd(options.EnableEtcd))
```
Comment (zyy17) [inline cmd/gtctl/cluster_create.go]: You don't have write it twice. You can also use the short one as following:

```
opts = append(opts, baremetal.WithEnableCache(options.EnableCache),  baremetal.WithEnableEtcd(options.EnableEtcd))
```

**Comment by daviderli614** [comment]
@confoc Change [CreateTableSQL](https://github.com/GreptimeTeam/gtctl/blob/develop/tests/e2e/greptimedbcluster_test.go#L35) to following can fixed the e2e failling:

```
CREATE TABLE dist_table (
                 ts TIMESTAMP DEFAULT current_timestamp(),
                 n INT,
    		row_id INT,
                TIME INDEX (ts),
                PRIMARY KEY(n)
)
 PARTITION ON COLUMNS (n) (
           n < 5,
           n >= 5 AND n < 9,
           n >= 9
)
```

**Comment by daviderli614** [comment]
Can add baremetal e2e test in next PR ?

**Comment by daviderli614** [comment]
Why close the PR ?

**Comment by confoc** [comment]
I don't know how I closed the pr.

**Comment by confoc** [comment]
I don't know how.Maybe I make some wrong operations.I don't remember, sry

**Comment by confoc** [comment]
>@daviderli614  Can add baremetal e2e test in next PR ?

I have already add baremetal e2e test in the PR

**Comment by daviderli614** [comment]
Seem ready review ? @confoc

File: cmd/gtctl/cluster_create.go [hunk 4]
```diff
@@ -62,6 +62,7 @@ type clusterCreateCliOptions struct {
 	Config             string
 	GreptimeBinVersion string
 	EnableCache        bool
+	metastore          bool
```
Comment (daviderli614) [inline cmd/gtctl/cluster_create.go]: suggestion: Use the upper name prifix, like `Metastore`.

File: pkg/cluster/baremetal/cluster.go [hunk 5]
```diff
@@ -34,6 +34,7 @@ type Cluster struct {
 	config       *config.BareMetalClusterConfig
 	createNoDirs bool
 	enableCache  bool
+	metastore    bool
```
Comment (daviderli614) [inline pkg/cluster/baremetal/cluster.go:37]: Use upper name prifix

File: cmd/gtctl/cluster_create.go (hunk 4 above)
Comment (confoc) [reply cmd/gtctl/cluster_create.go]: sure

File: cmd/gtctl/cluster_create.go [hunk 6]
```diff
@@ -62,6 +62,7 @@ type clusterCreateCliOptions struct {
 	Config             string
 	GreptimeBinVersion string
 	EnableCache        bool
+	Metastore          bool
```
Comment (zyy17) [inline cmd/gtctl/cluster_create.go]: I prefer the name `UseMemoryMeta`, which will be more accurate.

File: cmd/gtctl/cluster_create.go [hunk 7]
```diff
@@ -109,6 +110,7 @@ func NewCreateClusterCommand(l logger.Logger) *cobra.Command {
 	cmd.Flags().StringVar(&options.GreptimeDBClusterValuesFile, "greptimedb-cluster-values-file", "", "The values file for greptimedb cluster.")
 	cmd.Flags().StringVar(&options.EtcdClusterValuesFile, "etcd-cluster-values-file", "", "The values file for etcd cluster.")
 	cmd.Flags().StringVar(&options.GreptimeDBOperatorValuesFile, "greptimedb-operator-values-file", "", "The values file for greptimedb operator.")
+	cmd.Flags().BoolVarP(&options.Metastore, "memory-meta-storage", "m", false, "Bootstrap the whole cluster without installing etcd for testing purposes through using the memory storage of metasrv in bare-metal mode.")
```
Comment (zyy17) [inline cmd/gtctl/cluster_create.go]: ditto. If you use `UseMemoryMeta`, the long option should be `use-memory-meta`.

**Review by zyy17 — COMMENTED** [review]
Most of the PR are good for me.

**Review by zyy17 — COMMENTED** [review]
Most of the PR are good for me.

File: cmd/gtctl/cluster_create.go (hunk 7 above)
Comment (confoc) [reply cmd/gtctl/cluster_create.go]: Thanks for u good suggestion.I will commit a new pr soon

File: pkg/cluster/baremetal/cluster.go (hunk 5 above)
Comment (daviderli614) [inline pkg/cluster/baremetal/cluster.go]: suggestion: change to `useMemoryMeta`

**Review by zyy17 — APPROVED** [review]
lgtm
